build-system/tasks/check-types.js

```diff
@@ -49,7 +48,6 @@ async function checkTypes() {
 
   // Prepare build environment
   process.env.NODE_ENV = 'production';
-  cleanupBuildDir();
   typecheckNewServer();
```

Comment:
AFAICT the `debug` flag no longer has meaning in this file

Reply:
Should I remove? I don't know what debug/displayLifecycleDebugging are for.

Reply:
A quick inspection suggestions `displayLifecycleDebugging` should only matter if the `debug` method in that same file is called, and I don't see where it would be called here, but it's possible somewhere in the dependency chain it does get called. Not certain without testing it out

Reply:
`displayLifecycleDebugging` was added in #27700 as a way to print the output at various stages of compilation: babel, closure, more babel, etc. It's triggered by the `--debug` flag, and was previously used by `dist` and `check-types` (both of which used closure).

It's probably far less useful without closure and the pre- and post- babel steps. Okay to remove IMO. Might be worth checking in with someone like @jridgewell to make sure.